chain/network/src/peer_manager/peer_manager_actor.rs

```diff
@@ -392,22 +392,14 @@ impl PeerManagerActor {
                     // We belong to this edge.
                     if self.active_peers.contains_key(other) {
                         // This is an active connection.
-                        match edge.edge_type() {
-                            EdgeType::Removed => {
-                                self.maybe_remove_connected_peer(ctx, edge.clone(), other);
-                            }
-                            _ => {}
-                        }
-                    } else {
-                        match edge.edge_type() {
-                            EdgeType::Added => {
-                                // We are not connected to this peer, but routing table contains
-                                // information that we do. We should wait and remove that peer
-                                // from routing table
-                                self.wait_peer_or_remove(ctx, edge.clone());
-                            }
-                            _ => {}
+                        if edge.edge_type() == EdgeType::Removed {
+                            self.maybe_remove_connected_peer(ctx, edge.clone(), other);
                         }
+                    } else if edge.edge_type() == EdgeType::Added {
+                        // We are not connected to this peer, but routing table contains
+                        // information that we do. We should wait and remove that peer
+                        // from routing table
+                        self.wait_peer_or_remove(ctx, edge.clone());
                     }
```

Comment:
optional: another way to write such "decision tables" is to match on a tuple of conditions

```rust
let on_edge = self.active_peers.contains_key(other);
match (on_edge, edge.edge_type()) {
    (false, EdgeType::Added) => self.wait_peer_or_remove(ctx, edge.clone());
    (true, EdgeType::Removed) => self.maybe_remove_connected_peer(ctx, edge.clone(), other);
    (false, EdgeType::Removed) | (true, EdgeType::Added) => (),
}
```

The benefit is that it's more clear that you exhaustively check all combinations. The drawback is that, for many conditions, it might be hard to map from `(true, false, false)` pattern to what the conditions are.